### PR TITLE
ES6 classes, arrow functions, string templates

### LINKED ISSUE #6: Rewrite in ES6
Since Node >=4.0.0 supports ES6 classes, would be nice to rewrite the code using ES6 classes.
@izuzak If you agree, I can take care of that. :grin:

### PR DESCRIPTION
**Pull request by IonicaBizau**
- :art: Replaced the old class-like definition with ES6 `class` declaration.
- Used arrow functions making `self` variables obsolete.
- :fire: Removed the `version` field from the `module.exports` (will add that back if there is a good reason to keep it there). We can get the version from `package.json`.
- :memo: JSDoc comments and [`Blah`](https://github.com/IonicaBizau/blah) template to generate the `README.md` file.
- Generated the license via Blah––you will not need to take care of the license year anymore. Well, you will just run `blah -l license` once per year. :smile: 

To generate the `README.md` file we `blah -r` (or `blah --readme`). To know more about how to use `blah`, check out [this post I wrote](http://ionicabizau.net/blog/25-increasing-productivity-using-blah-2-0-0). :memo: 

Fixes #6. :tada: 

Ohh, and happy new year!!! :fireworks:

### CODE REVIEW

File: benchtable.js [hunk 1]
```diff
@@ -1,218 +1,235 @@
...
+     * @param {String} options Options object. A special boolean option `isTransposed`
+     * is available for Benchtable–if set to `true`, the output ASCII table will be
+     * transposed: the benchmarked functions will be organized into columns, while
+     * inputs will be organized into rows. This is useful for situations where the
+     * number of inputs is large, and the number of functions is small. This option
+     * defaults to `false`.
+     */
+    constructor (name, options) {
+
+        super(name, options);
```
Comment (IonicaBizau) [inline benchtable.js]: This currently doesn't work. But it will work after https://github.com/bestiejs/benchmark.js/pull/112 gets merged. :grin:

Comment (izuzak) [inline benchtable.js]: I think I see a new version of benchmark.js published on npm: https://www.npmjs.com/package/benchmark

@IonicaBizau do you want to bump the dependency version in this PR now? Once that's done, this is good to go, right? I haven't looked at it closely yet, will do that later today or tomorrow. Thanks again! :zap:

Comment (IonicaBizau) [inline benchtable.js]: > I think I see a new version of benchmark.js published on npm

@izuzak Great! I checked that when I woke up (6 hours ago) and saw that is not published. :smile:

Done in e6f85f86de7e56a1d6f01359dbf0d5da3586a86c and fixed the small issues after migration in the latest commits (replaced `pluck` with `map` in the example files and rebuilt the docs, use [`color-it`](https://github.com/ionicabizau/node-color-it) for colors).

Of course, check it out on your side and give me feedback. :grin:

File: README.md [hunk 2]
```diff
@@ -1,178 +1,229 @@
...
 
-**Specify functions to be benchmarked**
+#### Params
+- **Object** `config`: The options object. [See `benchmark.js` API docs.](http://benchmarkjs.com/docs#Suite_prototype_run)
 
-```javascript
-Benchtable.prototype.addFunction(name, func);
-```
+### `run(name, fun, options)`
+addFunction
```
Comment (izuzak) [inline README.md]: I think the above two lines should be merged into:

```
### `addFunction(name, fun, options)`
```

Right?

File: .blah/README.ejs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+```js
+<%- getExample("string_search_transposed.js") %>
+```
+
+## Contributing
+
+Want to contribute to this project? See information [here](CONTRIBUTING.md).
+
+## Credits
+Benchtable is developed by [Ivan Zuzak](http://ivanzuzak.info) \<izuzak@gmail.com\>;.
```
Comment (izuzak) [inline .blah/README.ejs]: I think there's an extra `;` at the end of the line.

File: README.md [hunk 4]
```diff
@@ -1,178 +1,229 @@
...
+// +----------------+--------------------+--------------------+-------------------+-----------------------+
+```
+
+## Transposed table
+
+If the Benchtable object is initialized with the `{isTransposed : true}` option (see [this example](/examples/string_search_transposed.js)), the script produces the same output but with rows and columns transposed:
 
-```javascript
+```js
 // import module
```
Comment (izuzak) [inline README.md]: I think this can be removed. Notice that the first line below is the same comment: https://github.com/izuzak/benchtable/pull/8/files#diff-04c6e90faac2675aa89e2176d2eec7d8R164

File: README.md [hunk 5]
```diff
@@ -1,178 +1,229 @@
...
-})
-.on('complete', function() {
-  console.log('Fastest is ' + this.filter('fastest').pluck('name'));
-  console.log(this.table.toString());
-})
-
-// run async
-.run({ 'async': false });
+// Import module
+var BenchTable = require('benchtable'); //require('benchtable');
```
Comment (izuzak) [inline README.md]: Just wondering -- why did you add the comment here? `//require('benchtable');`

File: README.md [hunk 6]
```diff
@@ -1,178 +1,229 @@
...
 
 This is the Benchtable version of the example from the [Benchmark.js homepage](http://benchmarkjs.com).
-Also available in the [examples directory](https://github.com/izuzak/benchtable/examples/string_search.js).
+Also available in the [examples directory](/examples/string_search.js).
+
+### Default behavior
+
+```js
+// Import module
+var BenchTable = require('benchtable'); //require('benchtable');
```
Comment (izuzak) [inline README.md]: What's the purpose of this comment? `//require('benchtable');`

**Comment by izuzak** [comment]
Other that those few minor comments -- looks good! Great work, @IonicaBizau -- thanks :boom:

File: README.md (hunk 2 above)
Comment (IonicaBizau) [inline README.md]: The jsdoc comment was wrong. Addressed in d0b1daa5dd57e2cdc8ad70abefeb894b472a7e33 and rebuilt README. :rocket:

File: .blah/README.ejs (hunk 3 above)
Comment (IonicaBizau) [inline .blah/README.ejs]: Fixed in dd5640c7d1f65aa54a5da9657fb706a6c3253b64. Btw, this looks weird in Atom! :boom: 

image

Before reporting this, can you double check if this is still an issue? I probably have an old version of Atom.

File: README.md (hunk 4 above)
Comment (IonicaBizau) [inline README.md]: :+1: 83d282d491bc886487ecb376f36b519aa9c408f7

File: README.md (hunk 5 above)
Comment (IonicaBizau) [inline README.md]: I didn't. `blah` added it. :smile: I removed it completely from example files in 83d282d491bc886487ecb376f36b519aa9c408f7.

File: README.md (hunk 6 above)
Comment (IonicaBizau) [inline README.md]: @izuzak It was there before (in the examples), and when running `blah -r` we [replace `../` with `benchtable`](https://github.com/izuzak/benchtable/commit/83d282d491bc886487ecb376f36b519aa9c408f7#diff-799839136677d9151042f5761071508cR17). :grin:

**Comment by IonicaBizau** [comment]
@izuzak Done! :joy: Probably you already noticed. I replaced `"foo".red` with [`colorIt("foo").red().toString()`](https://github.com/IonicaBizau/node-color-it/) because after updating the `cli-table` version we don't have these methods in the string prototype (which is good!), and because, why not–flat colors in terminal? :smile:

File: .blah/README.ejs (hunk 3 above)
Comment (IonicaBizau) [inline .blah/README.ejs]: There is already an issue regarding that: https://github.com/atom/markdown-preview/issues/320

File: README.md [hunk 7]
```diff
@@ -1,178 +1,203 @@
...
+
+```js
+// Import the benchtable module
+var BenchTable = require('benchtable');
+
+// Create benchtable suite
+var suite = new BenchTable();
+
+suite
+  benchtable/ Add functions for benchmarking
```
Comment (izuzak) [inline README.md]: I think there's something wrong with this line -- there's a single `/` and I think there should be two because it should be a comment, right? And it should be in a different place?

Comment (IonicaBizau) [inline README.md]: @izuzak Oh, great catch! The regex was replacing the things like crazy. Addressed it in https://github.com/IonicaBizauKitchen/benchtable/commit/1a00dea7dd630054ed9dae4ca3a820d2528597da#diff-799839136677d9151042f5761071508cR17 :tada:

**Comment by izuzak** [comment]
:sparkles: Thanks again for this, @IonicaBizau -- you're a :star2:!

**Comment by IonicaBizau** [comment]
@izuzak You're welcome! :blush: Glad to contribute. :tada:

I usually `npm i benchtable` when need to compare two functions and see which one is faster, since I stay all day in the terminal. :smile:
